app/models/container_image.rb

```diff
@@ -45,9 +47,10 @@ def operating_system=(value)
   end
 
   def docker_id
-    if image_ref.start_with?(DOCKER_IMAGE_PREFIX)
-      return image_ref[DOCKER_IMAGE_PREFIX.length..-1]
+    DOCKER_PREFIXES.each do |prefix|
+      return image_ref[prefix.length..-1] if image_ref.start_with?(prefix)
     end
```

Comment:
`valid_id?` seems like a function that will indicate if we have a validate the id in the object, not an outside id.
I am not sure why you are saying that we return two different things, we return the docker_id that we have. It changes in format depending on the docker version, but it is still the "docker_id".
I could make a function like `verify_id(id, repo_digests)` but it will just compare those values, and it will seem weird to separate them:
```
def verify_id(id, repo_digests)
  return true if id == docker_id
  repo_digests.each do |digest|
    return true if digest == docker_id
  end
  false
end
```

It seems like this logic should really be here (I would change the names to only actually call `docker_id` once).